File: Shell-Sort/python/leetcode_lc_506.py
score = [10,3,8,9,4]

class Solution:
    def findRelativeRanks(self, score: List[int]) -> List[str]:
        sorted = score[:];
        gap = int(len(sorted)/2);

        while gap > 0:

            for i in range(gap,len(sorted)):
                for j in range(i-gap,-1,-gap):
                    if sorted[j] < sorted[j+gap]:
                        sorted[j],sorted[j+gap] = sorted[j+gap],sorted[j];

            gap = int(gap/2);

        answer = []
        for item in score:
            v = sorted.index(item);
            if v == 0:
                answer.append("Gold Medal")
            elif v == 1:
                answer.append("Silver Medal")
            elif v == 2:
                answer.append("Bronze Medal")
            else:
                answer.append(str(v+1))

        return answer;


solution = Solution()
print(score)
print(solution.findRelativeRanks(score))

File: Shell-Sort/js/leetcode_lc_506.js
const score = [10,3,8,9,4]

/**
 * @param {number[]} score
 * @return {string[]}
 */
var findRelativeRanks = function(score) {
    const sorted = [...score]

    for(let gap = Math.floor(sorted.length/2) ; gap > 0 ; gap = Math.floor(gap/2)){
        for(let i = gap ; i < sorted.length ; i++){
            for(let j = i - gap ; j >= 0 ; j -= gap){
                if(sorted[j] < sorted[j+gap]){
                    [sorted[j],sorted[j+gap]] = [sorted[j+gap],sorted[j]]
                }
            }
        }
    }
    // console.log(sorted,score)
    const answer = score.map(item => {
        let v = sorted.indexOf(item)
        switch(v){
            case 0:
                v = "Gold Medal";
                break;
            case 1:
                v = "Silver Medal";
                break;
            case 2:
                v = "Bronze Medal";
                break;
            default:
                v = (v+1).toString()
        }
        return v;
    })
    return answer
};

console.log(score)
console.log(findRelativeRanks(score))